crown-court-proceeding/src/main/java/uk/gov/justice/laa/crime/crowncourt/service/ProceedingService.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.crowncourt.service;
+
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.justice.laa.crime.crowncourt.builder.UpdateRepOrderDTOBuilder;
+import uk.gov.justice.laa.crime.crowncourt.dto.CrownCourtDTO;
+import uk.gov.justice.laa.crime.crowncourt.model.ApiCrownCourtSummary;
+import uk.gov.justice.laa.crime.crowncourt.model.ApiProcessRepOrderResponse;
+import uk.gov.justice.laa.crime.crowncourt.staticdata.enums.CaseType;
+import uk.gov.justice.laa.crime.crowncourt.staticdata.enums.MagCourtOutcome;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.Set;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class ProceedingService {
+
+    private final RepOrderService repOrderService;
+    private final MaatCourtDataService maatCourtDataService;
+
+    private final Set<CaseType> caseTypes = Set.of(
+            CaseType.INDICTABLE, CaseType.CC_ALREADY, CaseType.APPEAL_CC, CaseType.COMMITAL
+    );
+
+    public ApiProcessRepOrderResponse processRepOrder(CrownCourtDTO dto) {
+        ApiProcessRepOrderResponse apiProcessCrownRepOrderResponse = new ApiProcessRepOrderResponse();
+        if (caseTypes.contains(dto.getCaseType()) ||
+                (dto.getCaseType() == CaseType.EITHER_WAY &&
+                        !dto.getMagCourtOutcome().equals(MagCourtOutcome.RESOLVED_IN_MAGS))) {
+            repOrderService.getRepDecision(dto);
+            repOrderService.determineCrownRepType(dto);
+            ApiCrownCourtSummary apiCrownCourtSummary = repOrderService.determineRepOrderDate(dto);
+            apiProcessCrownRepOrderResponse
+                    .withRepOrderDecision(apiCrownCourtSummary.getRepOrderDecision())
+                    .withRepOrderDate(apiCrownCourtSummary.getRepOrderDate())
+                    .withRepType(apiCrownCourtSummary.getRepType());
+        }
+        return apiProcessCrownRepOrderResponse;
+    }
+
+    public void updateApplication(CrownCourtDTO dto) {
+        processRepOrder(dto);
+        repOrderService.updateCCSentenceOrderDate(dto);
```

Comment:
Why do we need to update the repOrder twice, can we not do this as one update call ?